rosbag2_cpp/package.xml

```diff
@@ -10,7 +10,10 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
+  <build_depend>libboost-filesystem-dev</build_depend>
```

Comment:
this may be left from an old version of `master`? You'll probably want to revert all the changes from `rosbag2_cpp`, `rosbag2_storage_default_plugins`, `rosbag2_tests`, and `rosbag2_transport` - they don't seem related to your PR.

Additionally, I've noticed you checked in a `__pycache__` directory in `rosbag2_py` - that should be removed as well.